app-collab-note/src/components/Preview/index.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { useEffect, useState } from 'react';
+import { connect } from 'react-redux';
+import { render } from 'utils/MDUtils/mdToHtml';
+import { Resource } from 'utils/types';
+import './Preview.css';
+
+interface Props {
+  markdown: string;
+  resources: Resource[];
+}
+
+function Preview({ markdown, resources }: Props) {
+  const [html, setHtml] = useState('');
+
+  useEffect(() => {
+    setHtml(render(markdown, resources));
+  }, [markdown, resources]);
+
+  return (
+    <div dangerouslySetInnerHTML={{
+      __html: html
+    }}></div>
+  );
+}
+
+const mapStateToProps = (state: any) => {
```

Comment:
I would suggest creating an interface for the state, so that you don't need to use "any". It helps with code correctness, especially for a global state like this with many properties.

Reply:
Added AppState type and also refactored existing types.